packages/astro/src/content/runtime.ts

```diff
@@ -319,9 +334,84 @@ type RenderResult = {
 	remarkPluginFrontmatter: Record<string, any>;
 };
 
+const IMAGE_REGEX = /__ASTRO_IMAGE_="([^"]+)"/g;
+
+async function updateImageReferencesInBody(html: string, fileName: string) {
+	// @ts-expect-error Virtual module
+	const { default: imageAssetMap } = await import('astro:asset-imports');
+
+	const imageObjects = new Map<string, GetImageResult>();
+
+	// @ts-expect-error Virtual module resolved at runtime
+	const { getImage } = await import('astro:assets');
+
+	// First load all the images. This is done outside of the replaceAll
+	// function because getImage is async.
+	for (const [_full, imagePath] of html.matchAll(IMAGE_REGEX)) {
+		const decodedImagePath = JSON.parse(imagePath.replace(/&#x22;/g, '"'));
```

Comment:
A couple of things:
- the regex `/&#x22;/g` can be placed in a top-level constant; no need to create a new regex inside a loop. 
- `JSON.parse` *can fail* and we should handle the error

Reply:
I think this could actually just use `replaceAll` with a string. I nicked this from the Astro Image code, and probably missed something that needed it before.